libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2608,6 +2610,32 @@ Expression const* SMTEncoder::innermostTuple(Expression const& _expr)
 	return expr;
 }
 
+Expression const* SMTEncoder::cleanExpression(Expression const& _expr)
+{
+	auto const* expr = &_expr;
+	if (auto const* tuple = dynamic_cast<TupleExpression const*>(expr))
+		return cleanExpression(*innermostTuple(*tuple));
+	if (auto const* functionCall = dynamic_cast<FunctionCall const*>(expr))
+		if (
+			auto typeType = dynamic_cast<TypeType const*>(functionCall->expression().annotation().type);
+			typeType && *functionCall->annotation().kind == FunctionCallKind::TypeConversion
```

Comment:
Do we really need to check `typeType` if we know it's a TypeConversion?

Reply:
I was hoping this whole check could be simplified somehow. Maybe it is not necessary to check the result of the cast, but the cast itself is necessary, to extract the actual type.

Reply:
Right. How about only use the TypeConversion part here, and if that's true, assert that `typeType` is non null?

Reply:
That should work, I have updated the code according to this suggestion.